x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/execute_connector.ts

```diff
@@ -43,3 +44,75 @@ export function registerExecuteConnectorFunction({
     }
   );
 }
+
+export const GET_CONNECTOR_INFO_FUNCTION_NAME = 'get_connector_info';
+
+export function registerGetConnectorInfoFunction({
+  functions,
+  resources,
+}: FunctionRegistrationParameters) {
+  functions.registerFunction(
+    {
+      name: GET_CONNECTOR_INFO_FUNCTION_NAME,
+      description:
+        'Retrieves information about available connectors, including the parameter schema for each.',
+      visibility: FunctionVisibility.AssistantOnly,
+    },
+    async () => {
+      const actionsClient = await (
+        await resources.plugins.actions.start()
+      ).getActionsClientWithRequest(resources.request);
+
+      let allConnectors = await actionsClient.getAll();
+
+      // IDs of connectors to exclude (used for inference/AI purposes)
+      const excludedConnectorIds = new Set(['.bedrock', '.gemini', '.inference', '.gen-ai']);
+
+      // filter out AI connectors
+      allConnectors = allConnectors.filter(
+        (connector) => !excludedConnectorIds.has(connector.actionTypeId)
+      );
+
+      // get all connector parameters schemas
+      const connectorParamsSchemas = await Promise.all(
+        allConnectors.map(async (connector) => {
+          const actionType = await actionsClient.getActionType({
+            actionTypeId: connector.actionTypeId,
+          });
+          if (!actionType) {
+            return {
+              params: { type: 'object', properties: {} },
+              description: `Action type for connector with id ${connector.id} not found.`,
+            };
+          }
+          const paramsSchema = actionType.validate?.params?.schema;
+          if (!paramsSchema) {
+            return {
+              params: { type: 'object', properties: {} },
+              description: `Action type for connector with id ${connector.id} does not have a parameters schema.`,
+            };
+          }
+          return {
+            actionTypeId: connector.actionTypeId,
+            params: convertSchemaToOpenApi(paramsSchema),
+            description: `use this connector to ${actionType.description}` || '',
```

Comment:
Do you need the prefix?

```suggestion
            description: actionType.description,
```